src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -122,7 +122,8 @@ private static function compilePattern(Route $route, string $pattern, bool $isHo
 
         // Match all variables enclosed in "{}" and iterate over them. But we only want to match the innermost variable
         // in case of nested "{}", e.g. {foo{bar}}. This in ensured because \w does not match "{" or "}" itself.
-        preg_match_all('#\{(!)?(\w+)\}#', $pattern, $matches, \PREG_OFFSET_CAPTURE | \PREG_SET_ORDER);
+        $routeParamsPattern = $needsUtf8 ? '#\{(!)?([\p{L}_]+)\}#u' : '#\{(!)?(\w+)\}#';
```

Comment:
Numbers?

Reply:
Fixed, thanks

Reply:
I am not sure 100% sure, but \w with u flag should be enough. Am i wrong?

Reply:
\w doesn't support unicode characters.
Here is example:

- \w - https://regex101.com/r/p9O5Qd/1
- [\p{L}_] - https://regex101.com/r/QmU3Ru/1

Reply:
There is no unicode flag in your snippet